src/pages/resolve/pages/elections/ElectionSteps.vue

```diff
@@ -0,0 +1,299 @@
+                nominee => nominee.nominee_name === this.account
+            );
+            // this account already an active arbitrator
+            const foundUnavailableArbitrator = arbitrators.find(
+                arb => [1, 2].includes(arb) && arb.arb === this.account
+            );
+            return !foundNominee && !foundUnavailableArbitrator;
+        },
+        isRemoveNomineeButtonVisible() {
+            // needs to be in nominees table but not elections candidate
+            const {
+                resolve: { nominees }
+            } = this.$store.state;
+            const foundNominee = nominees.find(
+                nominee => nominee.nominee_name === this.account
+            );
+            const foundCandidate = this.currentElection.candidates.find(
+                candidate => candidate.name === this.account
+            );
+            if (foundNominee && !foundCandidate) return true;
+            return false;
+        },
+        isRemoveCandidacyButtonVisible() {
+            // needs to be in nominees table but not elections candidate
+            const foundCandidate = this.currentElection.candidates.find(
+                candidate => candidate.name === this.account
+            );
+            if (foundCandidate) return true;
+            return false;
+        },
+        isRegisterCandidateButtonVisible() {
+            const { resolve } = this.$store.state;
+            const { nominees } = resolve;
+            const { candidates } = this.currentElection;
+            const foundNominee = nominees.find(
+                nominee => nominee.nominee_name === this.account
+            );
+            const foundCandidate = candidates.find(
+                candidate => candidate.name === this.account
+            );
+            return !!this.account && foundNominee && !foundCandidate;
+        },
+        electionStatus() {
+            const { resolve } = this.$store.state;
+            if (resolve && resolve.config && resolve.elections) {
+                const {
+                    status,
+                    end_voting_ts,
+                    end_add_candidates_ts
+                } = this.currentElection;
+                if (status === 1) {
+                    const endAddCandidateUnixTimestamp = new Date(
+                        `${end_add_candidates_ts}Z`
+                    ).getTime();
+                    const rightNow = new Date().getTime();
+                    if (endAddCandidateUnixTimestamp > rightNow) {
+                        return 'candidate-registration';
+                    } else {
+                        return 'election-ready';
+                    }
+                }
+                if (status === 2) {
+                    // see if voting period has ended
+                    const endVotingUnixTimestamp = new Date(
+                        `${end_voting_ts}Z`
+                    ).getTime();
+                    const rightNow = new Date().getTime();
```

Comment:
`rightNow` can be assigned once  in parent scope, 272-273 and 260-262 can be refactored passing timestamp as var